include/client/job/job_bus.hpp

```diff
@@ -0,0 +1,42 @@
+#ifndef JOB_BUS_H
+#define JOB_BUS_H
+
+#include "client.hpp"
+#include "job_queue.hpp"
+
+#include <unordered_map>
+#include <functional>
+#include <queue>
+
+class JobBus{
+    typedef Job::Type Type;
+    
+    // Bool jobs handling
+    typedef std::function<void (std::string &t_arg, bool &t_ret)> b_JobsMethod;
+    typedef std::unordered_map<Type, b_JobsMethod> b_JobsMap;
+
+    // std::string job handling
+    typedef std::function<void (std::string &t_arg, std::string &t_ret)> s_JobsMethod;
+    typedef std::unordered_map<Type, s_JobsMethod> s_JobsMap;
+
+    // std::vector<std::string> job handling
+    typedef std::function<void (std::string &t_arg, std::vector<std::string> &t_ret)> v_JobsMethod;
+    typedef std::unordered_map<Type, v_JobsMethod> v_JobsMap;
```

Comment:
I don't think adding multiple maps to achieve polymorphism is the best way. can't we wrap this data in a struct so we can pass them around methods?

We could even make this struct from a virtual function so we can have different types being passed around.